diff --git a/archiso/mkarchiso b/archiso/mkarchiso
index 72a6e32..f0d7820 100755
--- a/archiso/mkarchiso
+++ b/archiso/mkarchiso
@@ -166,14 +166,6 @@ _pacman ()
 _cleanup () {
     _msg_info "Cleaning up what we can on airootfs..."
 
-    # Delete initcpio image(s)
-    if [[ -d "${work_dir}/airootfs/boot" ]]; then
-        find "${work_dir}/airootfs/boot" -type f -name '*.img' -delete
-    fi
-    # Delete kernel(s)
-    if [[ -d "${work_dir}/airootfs/boot" ]]; then
-        find "${work_dir}/airootfs/boot" -type f -name 'vmlinuz*' -delete
-    fi
     # Delete pacman database sync cache files (*.tar.gz)
     if [[ -d "${work_dir}/airootfs/var/lib/pacman" ]]; then
         find "${work_dir}/airootfs/var/lib/pacman" -maxdepth 1 -type f -delete
